src/notification/index.js

```diff
@@ -71,19 +71,14 @@ export const getAccountFromNotificationData = (
   }
 
   if (urlMatches.length > 1) {
-    // The user has several accounts in the notification's realm.  We should
-    // be able to tell the right one using the notification's `user_id`...
-    // except we don't store user IDs in `accounts`, only emails.  Until we
-    // fix that, just ignore the information.
-    logging.warn('notification realm_uri ambiguous; multiple matches found', {
-      realm_uri,
-      parsed_url: realmUrl,
-      match_count: urlMatches.length,
-      unique_identities_count: new Set(urlMatches.map(matchIndex => identities[matchIndex].email))
-        .size,
-    });
-    // TODO get user_id into accounts data, and use that
-    return null;
+    // The user has several accounts in the notification's realm. Pick one
+    // based on the notification's `user_id`.
+    const userMatch = urlMatches.find(urlMatch => accounts[urlMatch].userId === user_id);
+    if (userMatch == null) {
+      logging.warn('notification account not found with realm_uri and user_id');
+      return null;
+    }
+    return userMatch;
   }
 
   return urlMatches[0];
```

Comment:
Let's do the search for a user_id match regardless of whether there are one or several realm-URL matches. That way we notice if the one account in the realm isn't actually for the right user.